provider/pkg/await/deployment.go

```diff
@@ -689,7 +712,23 @@ func (dia *deploymentInitAwaiter) changeTriggeredRollout() bool {
 		return true
 	}
 
-	return dia.deployment.GetAnnotations()[revision] != dia.config.lastOutputs.GetAnnotations()[revision]
+	// There is a race condition where the deployment object's revision and status has not been updated yet. We have to keep polling
+	// until the status is updated to ensure that the deployment controller has started the rollout. This function is called from
+	// within the replicaset event watcher code, and the deployment object is updated by the deployment event watcher code, which is why
+	// we also need to do this check here as well to ensure that the deployment object has been updated from the event log.
+	for {
+		generation := dia.deployment.GetGeneration()
+		observedGeneration, _, _ := unstructured.NestedInt64(dia.deployment.Object, "status", "observedGeneration")
+
+		if generation == observedGeneration {
```

Comment:
Related to my comment above, I don't _think_ we need this polling if we ignore replica events until the deployment has started -- but this code is gnarly and there's a good chance I'm wrong!

Reply:
Agreed, the await logic is somewhat gnarly!